src/api/Workspace.ts

```diff
@@ -1,24 +1,61 @@
 import { StartRequest } from './methods/start';
-import { ServiceRequest, ServiceResponse } from './methods/service';
-import { RegisterRequest } from './methods/register';
+import { ServicesMap, ServicesRequest } from './methods/services';
+import { RegisterServiceRequest } from './methods/registerService';
+import { ComponentsMap, ComponentsRequest } from './methods/components';
+import { ConfigRequest } from './methods/config';
 
-interface Workspace {
-  // Load all the services from url
-  // Load the layout and render components inside
+/**
+ * Workspace is the core service of CapsulaHub, it is responsible for :
+ * - Loading services and components that are included in its configuration (START)
+ * - Allowing them to register themselves (REGISTER)
+ * - Letting them communicate together (SERVICES/COMPONENTS)
+ * - Getting their own configuration (CONFIG)
+ */
+export interface Workspace {
+  /**
+   * Loading all services and components included in workspace configuration
+   * Reject in case :
+   * - Workspace already started
+   * - Workspace start already in progress
+   * - An error with importing a service occurred
+   * - An error with importing a component occurred
+   * - An error with registering a component occurred
+   * @param startRequest
+   */
   start(startRequest: StartRequest): Promise<void>;
 
-  // Returns proxy of a registered service
-  // Can be used by any service during run time to get a proxy of another service
-  service(serviceRequest: ServiceRequest): Promise<ServiceResponse>;
+  /**
+   * Getting promises to each service that has been loaded in the workspace
+   * Reject in case :
+   * - Workspace not started yet
+   * @param servicesRequest
+   */
+  services(servicesRequest: ServicesRequest): Promise<ServicesMap>;
 
-  // Register a service or a component in the workspace
-  // Can be used by any service to register itself and become available with Workspace.service method
-  register(registerRequest: RegisterRequest): Promise<void>;
+  /**
+   * Getting promises to each component that has been loaded in the workspace
+   * Reject in case :
+   * - Workspace not started yet
+   * @param componentsRequest
+   */
+  components(componentsRequest: ComponentsRequest): Promise<ComponentsMap>;
 
-  /*
-  // For future / To be defined
-  // Allow persistence of services state
-  persist({ serviceName: string }): Promise<>;
-  state({ serviceName: string }): Observable<ServiceState>;
-  */
+  /**
+   * Register a service in the workspace.
+   * Reject in case :
+   * - Workspace not started yet
+   * - Service specified in request doesn't exist in workspace configuration
+   * - Service specified in request already registered
+   * - Invalid request
+   * @param registerServiceRequest
+   */
+  registerService(registerServiceRequest: RegisterServiceRequest): Promise<void>;
```

Comment:
I thought that all services and components, that we have in configuration, will be registered, when Workspace is started.
I this case why do we need this method to be public?
If we have registerService - what about registerComponent?

Reply:
`registerComponent` is internal and will be called by the workspace itself in start method implementation.
`registerService` is public because it will be called by external services in their bootstrap code.